share/functions/__fish_print_hostnames.fish

```diff
@@ -5,10 +5,11 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
 	if type -q getent
 		getent hosts 2>/dev/null | tr -s ' ' ' ' | cut -d ' ' -f 2- | tr ' ' '\n'
 	else if test -r /etc/hosts
-		tr -s ' \t' '  ' < /etc/hosts | sed 's/ *#.*//' | cut -s -d ' ' -f 2- | __fish_sgrep -o '[^ ]*'
+		tr -s ' \t' '  ' < /etc/hosts | sed 's/ *#.*//' | sed 's/^[0.]* .*$//' | cut -s -d ' ' -f 2- | __fish_sgrep -o '[^ ]*'
```

Comment:
Why sed and not grep (`grep -v '^0.0.0.0'`)?

Also, on my system the `getent` path also prints these.

And while you are at it, you might also replace some of these commands with `string`.